app/models/util/period.rb

```diff
@@ -130,14 +130,32 @@ def with(start_date = Time.zone.today, end_date = Time.zone.today, label = nil)
     def parse_date(date)
       return nil if date.blank?
 
-      if date.is_a? String
+      if date.is_a?(String)
+        if date.match?(/\A\d{1,2}\.\d{1,2}\.\d{2}\z/)
+          day, month, short_year = date.split('.').map(&:to_i)
+
+          # Get current full year and its century
+          current_year = Time.zone.today.year
+          cutoff = (current_year + 5) % 100
+          current_century = (current_year / 100) * 100
+
+          full_year = if short_year <= cutoff
+                        current_century + short_year
+                      else
+                        (current_century - 100) + short_year
+                      end
+
+          return Date.new(full_year, month, day)
+        end
+
         begin
           date = Date.strptime(date, I18n.t('date.formats.default'))
-        rescue StandardError
+        rescue ArgumentError
           date = Date.parse(date)
         end
       end
-      date = date.to_date if date.is_a? Time
+
+      date = date.to_date if date.is_a?(Time)
       date
     end
```

Comment:
`strptime` can handle the parsing of two year dates. Locale specific formats are handled in Rails with the I18n gem.

I'd rewrite this method as:

```ruby
    def parse_date(date)
      return nil if date.blank?

      case date
      when Time then date.to_date
      when /\A\d{1,2}\.\d{1,2}\.\d{2}\z/ then parse_date_string(date, :short_year)
      when String then parse_date_string(date)
      else date
      end
    end

    private

    def parse_date_string(date, type=:default)
      Date.strptime(date, I18n.t(type, scope: 'date.formats'))
    rescue ArgumentError
      Date.parse(date)
    end
...
```
and add the following to `config/locales/crud.de-CH.yml`:
```YAML
de-CH:
  ...
  # formats
  date:
    formats:
      short_year: "%d.%m.%y"
```